Extension/src/LanguageServer/configurations.ts

```diff
@@ -2030,6 +2007,43 @@ export class CppProperties {
             const pattern: RegExp = new RegExp(`"[^"]*?(?<="|;)${escapedPath}(?="|;).*?"`, "g");
             const configMatches: string[] | null = curText.match(pattern);
 
+            const expandedPaths: string[] = this.resolveAndSplit([curPath], undefined, this.ExtendedEnvironment, true, false);
+            const incorrectExpandedPaths: string[] = [];
+
+            if (expandedPaths.length > 0) {
+                if (this.rootUri) {
+                    for (const [index, expandedPath] of expandedPaths.entries()) {
+
+                        if (expandedPath.includes("${workspaceFolder}")) {
+                            expandedPaths[index] = this.resolvePath(expandedPath, false);
+                        } else {
+                            expandedPaths[index] = this.resolvePath(expandedPath);
+                        }
+
+                        const checkPathExists: any = util.checkPathExistsSync(expandedPaths[index], this.rootUri.fsPath + path.sep, isWindows, false);
+                        if (!checkPathExists.pathExists) {
+                            // If there are multiple paths, store any non-existing paths to squiggle later on.
+                            incorrectExpandedPaths.push(expandedPaths[index]);
+                        }
+                    }
+                }
+            } else {
+                continue;
+            }
+
+            const pathExists: boolean = incorrectExpandedPaths.length === 0;
+
+            for (const [index, expandedPath] of expandedPaths.entries()) {
+                // Normalize path separators.
+                if (path.sep === "/" && expandedPaths.length > 0) {
```

Comment:
This condition seems incorrect. Should it be

```
if (expandedPaths.length > 0) {
    if (path.sep === "/") {
    ...
    } else {
    ...
    }
}
```

Reply:
umm...wait, I think you can get rid of the `expandedPaths.length > 0` checked because it won't be hit since it's iterating over expandedPaths.entries().

Reply:
Well...is the check supposed to be `expandedPath.length > 0`?

Reply:
The expandedPath.length > 0 check should have been deleted when I added the for loop